### DISCUSSION TITLE
Newbie question about Chrome updates

### DISCUSSION THREAD

**Discussion by Kimble303** [FlareSolverr/FlareSolverr#1023] [Q&A] [answered]
I use Windows 11 and I have Jackett running as a background service. I don't have Chrome browser installed.

I downloaded the latest FlareSolverr release "flaresolverr_windows_x64.zip" from GitHub, extracted the contents (including chrome-directory) and run the flaresolverr.exe. No problem with terminal, everything is OK and FlareSolverr is running successfully with Jackett (with FlareSolverr API URL configured).

FlareSolverr uses Chromium 115? Right? Is it possible to somehow manually update the version (= update the contents of the chrome-directory of FlareSolverr)? Or how do you update just Chrome version? Is that even possible? Or is it so that the chrome-directory can be updated only when you download the new (latest) release of FlareSolverr, extract the contents of the zip (including chrome-directory) and replace the old files?

Or should I install Chrome browser?

**Comment by ilike2burnthing** [comment, marked as answer]
https://commondatastorage.googleapis.com/chromium-browser-snapshots/index.html?prefix=Win_x64/

Sort by `Last Modified`, wait for `LAST_CHANGE` to load, click `LAST_CHANGE`, open the downloaded .txt file, copy the version number, paste that into the `Filter` box on the webpage, open that directory, download `chrome-win.zip`, delete the contents of the `chrome` folder in your FlareSolverr directory, extract the contents of `chrome-win.zip` to the now empty `chrome` folder, run FlareSolverr.

At the time of writing the version number is `1240665`, which gives you `v122.0.6203.0` of Chromium.

**Reply by Kimble303** [reply]
Thanks for the info!

I already managed to download the latest chromium snapshot earlier but hesitated to try and update because I thought there might be some problems if you just copy/paste files instead of performing a proper install procedure.

But seemingly there won't be any problem at all. 

Thanks once again!

**Comment by ilike2burnthing** [comment]
This can be automated with the following batch script (either ran manually or as a scheduled task):

```cmd
@ECHO OFF
set URL=https://www.googleapis.com/download/storage/v1/b/chromium-browser-snapshots/o/Win_x64
set FILE=chrome-win.zip?alt=media
for /F "delims=" %%i in (LATEST_CHANGE) do set LATEST_CHANGE=%%i

echo --------------------------------------------------------------------------------
echo I                                                                              I
echo I                         Checking for new version ...                         I
echo I                                                                              I
echo --------------------------------------------------------------------------------

wget --no-check-certificate %URL%%%2FLAST_CHANGE?alt=media
ren "Win_x64%%2FLAST_CHANGE@alt=media" CHECK
for /F "delims=" %%j in (CHECK) do set CHECK=%%j
if not "%LATEST_CHANGE%"=="%CHECK%" goto Download

echo --------------------------------------------------------------------------------
echo I                                                                              I
echo I             You have the latest Chromium version - Build %LATEST_CHANGE%.            I
echo I                                                                              I
echo --------------------------------------------------------------------------------

del CHECK

goto eof

:Download

echo --------------------------------------------------------------------------------
echo I                                                                              I
echo I           Downloading latest version of Chromium Build %CHECK% ...           I
echo I                                                                              I
echo --------------------------------------------------------------------------------

wget --no-check-certificate %URL%%%2F%CHECK%%%2F%FILE%

echo --------------------------------------------------------------------------------
echo I                                                                              I
echo I          Updating Chromium from Build %LATEST_CHANGE% to Build %CHECK% ...           I
echo I                                                                              I
echo --------------------------------------------------------------------------------

taskkill /IM flaresolverr.exe /F
if exist "chrome-win.zip" del chrome-win.zip
ren "*chrome-win.zip*" chrome-win.zip
rmdir /s /q chrome
7za x chrome-win.zip
ren chrome-win chrome
move /y CHECK LATEST_CHANGE
start flaresolverr /c

echo --------------------------------------------------------------------------------
echo I                                                                              I
echo I                Update successful. Now running Build %CHECK%.                 I
echo I                                                                              I
echo --------------------------------------------------------------------------------

:eof

timeout 5
```

Expects to be run within the root of the FlareSolverr directory, along with `wget.exe` and `7za.exe`:
- https://eternallybored.org/misc/wget/ (currently [v1.21.4 x64 EXE](https://eternallybored.org/misc/wget/1.21.4/64/wget.exe))
- https://www.7-zip.org/download.html (currently [v23.01 7-Zip Extra .7z](https://www.7-zip.org/a/7z2301-extra.7z) - extract 7za.exe from it)

**Comment by ilike2burnthing** [comment]
I should add, we'll update it eventually, but as of yet there haven't been any issues.

**Reply by Kimble303** [reply]
Somehow running the script was not a success. I think there is a simple explanation but I am too stupid to figure it out:

```
The system cannot find the file LATEST_CHANGE.
--------------------------------------------------------------------------------
I                                                                              I
I                         Checking for new version ...                         I
I                                                                              I
--------------------------------------------------------------------------------
--2023-12-24 01:00:37--  https://www.googleapis.com/download/storage/v1/b/chromium-browser-snapshots/o/Win_x64%2FLAST_CHANGE?alt=media
Resolving www.googleapis.com (www.googleapis.com)... 74.125.131.95, 108.177.14.95, 142.250.150.95, ...
Connecting to www.googleapis.com (www.googleapis.com)|74.125.131.95|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 7 [text/plain]
Win_x64%2FLAST_CHANGE@alt=media: Permission denied

Cannot write to 'Win_x64%2FLAST_CHANGE@alt=media' (Bad file descriptor).
The system cannot find the file specified.
The system cannot find the file CHECK.
--------------------------------------------------------------------------------
I                                                                              I
I             You have the latest Chromium version - Build .            I
I                                                                              I
--------------------------------------------------------------------------------
Could Not Find C:\Program Files\Flaresolverr\flaresolverr_windows_x64\flaresolverr\CHECK

Waiting for 2 seconds, press a key to continue ...
```

**Reply by ilike2burnthing** [reply]
Don't run FlareSolverr (or this script) out of `Program Files`, use `ProgramData` like Jackett.

**Reply by Kimble303** [reply]
Thanks again!

ProgramData directory works nicely! 

Shoulda guesstimated that using Program Files directory may cause some issues... *sigh*
